core/backend/src/test/integration/PushChanges.test.ts

```diff
@@ -132,17 +131,17 @@ describe("PushChangesTest (#integration)", () => {
       iModel.saveChanges();
 
       // Set the token to expire four minutes from now
-      const jwt = requestContext.accessToken;
-      const fourMinFromNow = new Date(Date.now() + 2 * 60 * 1000);
-      const expiringToken = new AccessToken(jwt.toTokenString(IncludePrefix.No), jwt.getStartsAt(), fourMinFromNow, jwt.getUserInfo());
-      const expiringContext = new AuthorizedClientRequestContext(expiringToken);
+      // const jwt = requestContext.accessToken;

```

Comment:
Is this no longer refreshing the tokens?

Reply:
This is something I'm a little confused on. Since the Access token no longer stores the expiration, what mechanism in this is checking the expiration and whether it should refresh or not? Also not sure how to mimic that here since we are only passing the token in now.

Reply:
I also don't understand what's refreshing the token here.
I was under the impression that token refresh happens behind the scenes within an `AuthorizationClient` and the refreshed tokens only bubble up when something else requests a new token from the `AuthorizationClient`. Unless something within `BriefcaseDb.pushChanges` is supposed to refresh the token and I can't find that part?

When receiving a token response from the token issuing authority, the expiration time should always be packaged with the token string, so the `AuthorizationClient` should have a straightforward way to just cache that. If expiration needs to be known, the `AuthorizationClient` instance could then be consulted for the expiration date.
Alternatively, if we really need the expiration on hand, the request context could optionally hold that information alongside the token string in a composite object (although that just sounds like circling back to the old `AccessToken` class to me).